### PR TITLE
new feature: ch-nudge

### PR DESCRIPTION
**Pull request by j-ogas**
Addresses: #740.

### CODE REVIEW

File: doc/ch-nudge_desc.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+underlying code with :code:`ch-grow(1)`.
+
+:code:`ACTION` must be one of the following:
+
+  :code:`delete IMAGE_REF`
+    TODO
+  :code:`list IMAGE_REF`
+    TODO
+  :code:`push IMAGE_REF`
+    TODO
```
Comment (reidpr) [inline doc/ch-nudge_desc.rst:39]: I wonder if we should also consider this interface:

```
$ ch-nudge --delete IMAGE_REF
$ ch-nudge --list IMAGE_REF
$ ch-nudge IMAGE_REF [DEST_IMAGE_REF]
```

i.e.:

1. default is push
2. would need to document that list should not have a tag or hash

File: doc/ch-nudge_desc.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Other arguments:
+
+  :code:`-h`, :code:`--help`
+    Print help and exit.
+
+  :code:`--chunk SIZE`
+    Upload layer in chunks of size :code:`SIZE`. This is typically unnecessary
+    and should be used only by users who are familiar with image uploading and
+    why this may useful.
```
Comment (reidpr) [inline doc/ch-nudge_desc.rst:51]: Would we ever need this? If not, we should leave it out.

File: doc/ch-nudge_desc.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Examples
+========
+
+Push image "hello-world" to the Charliecloud image repository on the Docker
+registry.
+
+::
+
+  $ ch-nudge push registry-1.docker.io:443/charliecloud/hello-world
+  pushing image: hello-world
```
Comment (reidpr) [inline doc/ch-nudge_desc.rst:79]: I'd print the full image ref being used here, like `ch-tug` does.

File: doc/ch-nudge_desc.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  deleting image: hello-world
+  [...]
+
+List all available tags in the centos repository on the Docker registry.
+
+::
+
+  $ ch-nudge list registry-1.docker.io:443/charliecloud/tags/list
+  listing tags
+  [...]
```
Comment (reidpr) [inline doc/ch-nudge_desc.rst:97]: For this one I think example output would be especially good. I do wonder about more complete output for all the examples, though I think the way to put that in is copy-and-paste later.

File: doc/ch-nudge_desc.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+List all available repositories from the Docker registry.
+
+::
+
+  $ ch-nudge list registry-1.docker.io:443/_catalog
```
Comment (reidpr) [inline doc/ch-nudge_desc.rst:103]: The `_catalog` is pretty inside baseball. If we have list capabilities at all, then the syntax here I'd suggest something like:

```
$ ch-nudge list registry-1.docker.io:443
```

That said, who actually knows what `registry-1.docker.io` means?

Is listing by browsing the registry website adequate?

**Comment by j-ogas** [comment]
Closing in favor of #867.

**Pull request closed without merging**